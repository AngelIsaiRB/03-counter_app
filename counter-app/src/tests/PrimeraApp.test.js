import React from 'react';
import  { shallow } from 'enzyme';

// const { render } = require("@testing-library/react");
const { default: PrimeraApp } = require("../PrimeraApp");

describe('pruebas en  <primeraAPP/> ', () => {
    
    // test('should mostrar el mensaje "Hola,soy isai "', () => {

    //     const saludo = "hola, soy isai!";
    //     const {getByText} = render(<PrimeraApp saludos={saludo}/>);

    //     expect(getByText(saludo)).toBeInTheDocument();

    // });

    test('should de mostrar <primeraApp/> correctamente', () => {

        const saludo = "hola, soy ";
        const wrapper = shallow(<PrimeraApp saludos={saludo}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should mostrar el subtitulo enviado por props', () => {
         
        const saludo = "hola, soy goku";
        const subTitulo = "hola, soy una subtitulo";
        const wrapper = shallow(<PrimeraApp saludos={saludo} subtitulo={subTitulo}/>);
        const textoParrafo = wrapper.find('p').text();        
        expect(textoParrafo).toBe(subTitulo);

    });
    
    
    
});
