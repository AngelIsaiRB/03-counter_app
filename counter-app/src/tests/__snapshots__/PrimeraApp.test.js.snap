// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en  <primeraAPP/>  should de mostrar <primeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
     
    hola, soy 
     
  </h1>
  <p>
    Soy un subtitutlo
  </p>
</Fragment>
`;
